File: hw_2_layouts/src/main/java/com/gmail/elnora/fet/hw_2_layouts/MainActivity.java
package com.gmail.elnora.fet.hw_2_layouts;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.Toast;
import java.util.Locale;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.task6_custom_view);

        CustomView customView = findViewById(R.id.colorCircle);
        customView.setListener(new CustomView.OnTouchEvent() {
            @Override
            public void onTouchEvent(float eventX, float eventY) {
                String message = String.format(Locale.getDefault(), "Нажаты координаты [%.0f;%.0f]", eventX, eventY);
                Toast.makeText(MainActivity.this, message, Toast.LENGTH_SHORT).show();
            }
        });
    }

}

File: hw_8_weather_app/src/main/java/com/gmail/elnora/fet/hw_8_weather_app/MainActivity.kt
package com.gmail.elnora.fet.hw_8_weather_app

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        showMainFragmentWeather(savedInstanceState)
    }

    private fun showMainFragmentWeather(savedInstanceState: Bundle?) {
        if (findViewById<View?>(R.id.fragmentContainer) != null) {
            if (savedInstanceState != null) {
                return
            }
            showFragment(WeatherFragment.newInstance(), WeatherFragment.TAG)
        }
    }

    private fun showFragment(fragment: Fragment, tag: String) {
        supportFragmentManager.beginTransaction()
                .add(R.id.fragmentContainer, fragment, tag)
                .commit()
    }
}
